src/GraphQL/Authorization/AuthorizationExtensions.cs

```diff
@@ -63,6 +80,39 @@ public static TMetadataProvider AuthorizeWith<TMetadataProvider>(this TMetadataP
             return provider;
         }
 
+        /// <summary>
+        /// Adds authorization role(s) to the specified metadata provider. Roles should
+        /// be comma-separated and role names will be trimmed. If the underlying field already
+        /// contains a role with the same name, then it will not be added twice.
+        /// </summary>
+        /// <typeparam name="TMetadataProvider"> The type of metadata provider. Generics are used here to
+        /// let compiler infer the returning type to allow methods chaining.
+        /// </typeparam>
+        /// <param name="provider">
+        /// Metadata provider. This can be an instance of <see cref="GraphType"/>,
+        /// <see cref="FieldType"/>, <see cref="Schema"/> or others.
+        /// </param>
+        /// <param name="roles"> Comma-separated list of authorization role name(s). </param>
+        /// <returns> The reference to the specified <paramref name="provider"/>. </returns>
+        public static TMetadataProvider AuthorizeWithRoles<TMetadataProvider>(this TMetadataProvider provider, string roles)
```

Comment:
Add overload with `params string[] roles`? I'm a bit confused here about difference between _policy_ and _role_ and about chaining design:
```csharp
field.AuthorizeWith("mypolicy1").AuthorizeWith("mypolicy2");
```
but
```csharp
field.AuthorizeWithRoles("myrole1,myrole2").AuthorizeWithRoles("myrole3;myrole4;myrole5");
```

I do not like that in one case, a single number is used, and in another multiple. Techically we can implement both methods to accept multiple policies/roles. I would like to see clean API design here.

Reply:
Well typically you either authorize with a single policy, or a number of allowable roles.

I have no problem changing `AuthorizeWithRoles()` to use either params, or an overload for both params and comma-separated.  But I feel that GraphQLAuthorizeAttribute should match the ASP.Net Core attribute.  And since it does, I would leave the comma-separated overload.

So I'll quickly add overloads for params to `AuthorizeWithRoles()`.  There should be no need to add `AuthorizeWithPolicies(params string[])` -- this would be an edge case and the user can simply call `AuthorizeWithPolicy` twice if necessary.

Reply:
> typically you either authorize with a single policy, or a number of allowable roles.

I think so too. Worth to mention that in xml comments.

> I feel that GraphQLAuthorizeAttribute should match the ASP.Net Core attribute

OK, I'm talking here more about extension methods for graphtype first approach.

>  There should be no need to add AuthorizeWithPolicies(params string[])

OK.